Libraries/Components/FocusableWindows/FocusableWindows.windows.js

```diff
@@ -0,0 +1,271 @@
+ *     render () {
+ *
+ *         styleSet = FocusableWindows.splitStyle(this.prop.styles);
+ *
+ *         return (
+ *            <FocusableWindows
+ *               ref = {(f) => this._focusable = f}
+ *               style = {styleSet.focusableStyle}
+ *               onKeyDown = {...}
+ *               handledKeyDownKeys: {[FocusableWindows.keys.Enter]}
+ *            >
+ *              <SomeComponent
+ *                  {...this.props}
+ *                  ref = {(c) => this._child = c}
+ *                  style = {styleSet.childStyle}
+ *              />
+ *            </FocusableWindows>
+ *         );
+ *      }
+ *
+ *      // important for animations
+ *      setNativeProps (props) {
+ *          var splitProps = FocusableWindows.splitNativeProps(props);
+ *          if (splitProps.focusableProps !== undefined) {
+ *             this._focusable.setNativeProps(splitProps.focusableProps)
+ *          }
+ *          if (splitProps.childProps !== undefined) {
+ *             this._child.setNativeProps(splitProps.childProps);
+ *          }
+ *      }
+ *   }
+ *
+ * @providesModule FocusableWindows
+ */
+'use strict';
+
+var PropTypes = require('prop-types');
+var React = require('React');
+var ReactNative = require('ReactNative');
+var ViewPropTypes = require('ViewPropTypes');
+var requireNativeComponent = require('requireNativeComponent');
+var UIManager = require('UIManager');
+const flattenStyle = require('flattenStyle');
+
+class FocusableWindows extends React.Component {
+
+  static propTypes = {
+    ...ViewPropTypes,
+
+    /**
+     * Controls whether the view is a tab stop. Useful for buttons and other
+     * controls that can be focused.
+     *
+     * @platform windows
+     */
+    isTabStop: PropTypes.bool,
+
+    /**
+     * Sets the tab index for the view.
+     *
+     * @platform windows
+     */
+    tabIndex: PropTypes.number,
+
+    /**
+     * Sets the keyboard navigation mode for the view.
+     *
+     * @platform windows
+     */
+    tabNavigation: PropTypes.oneOf([
+      'local',
+      'cycle',
+      'once'
+    ]),
+
+    /**
+     * Controls whether control should use system default provided focus rects
+     *
+     * @platform windows
+     */
+    disableSystemFocusVisuals: PropTypes.bool,
+
+    /**
+     * Called when the view receives focus.
+     *
+     * @platform windows
+     */
+    onFocus: PropTypes.func,
+
+    /**
+     * Called when the view focus is lost.
+     *
+     * @platform windows
+     */
+    onBlur: PropTypes.func,
+
+    /**
+     * Set of keys that should be handled on key down by this component.
+     *
+     * @platform windows
+     */
+    handledKeyDownKeys: PropTypes.arrayOf(PropTypes.number),
+
+    /**
+     * Set of keys that should be handled on key up by this component.
+     *
+     * @platform windows
+     */
+    handledKeyUpKeys: PropTypes.arrayOf(PropTypes.number),
+
+    /**
+     * Called when key down while component has focus.
```

Comment:
>has focus [](start = 44, length = 9)

Can this also fire if a descendant has focus?

Reply:
Yes, topFocus/Blur and topKeyDown/Up are bubbling events (on React side), so depending on what handlers called (or not) stopPropagation, you may get them here as well.

---
In reply to: [157296675](https://github.com/Microsoft/react-native-windows/pull/1546#discussion_r157296675) [](ancestors = 157296675)